app/helpers/stream_entries_helper.rb

```diff
@@ -48,22 +48,25 @@ def microformats_h_class(status, is_predecessor, is_successor, include_threads)
   end
 
   def rtl?(text)
-    rtl_characters = /[\p{Hebrew}|\p{Arabic}|\p{Syriac}|\p{Thaana}|\p{Nko}]+/m.match(text)
+    rtl_charcount = text.gsub(/[^\p{Hebrew}|\p{Arabic}|\p{Syriac}|\p{Thaana}|\p{Nko}]/m, '').strip.size.to_f
 
-    if rtl_characters.present?
-      total_size = text.strip.size.to_f
-      rtl_size(rtl_characters.to_a) / total_size > 0.3
+    if rtl_charcount.positive?
+      # Remove mentions before counting characters to decide RTL ratio
+      justtext = text.gsub(Account::MENTION_RE, '')
+      # Remove Email addresses
+      justtext = justtext.gsub(/^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$/, '')
+      # Naiive catcher for URLs
+      justtext = justtext.gsub(/[-A-Za-z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-A-Za-z0-9@:%_\+.~#?&\/=]*)/, '')
```

Comment:
just to make a point, I added a language detection spec, here's what I got:

>   1) LanguageDetector to_iso_s Detects Hebrew language
>     Failure/Error: expect(result).to eq :he
>       expected: :he
>          got: :da
>     (compared using ==)

Yes, my Hebrew+mentions+hashtag is danish. Sorry, I can't rely on this mechainsm for directonality. if it gets fixed or improved, it can be retrofitted later for this task, but for now please merge this into 1.4.